### ISSUE TITLE
Incorrect Savitzky-Golay coefficients in scipy.signal

### ISSUE THREAD

**Issue by RutgerK** [scipy/scipy#7209]
Using Scipy 0.18.1 installed with Conda on Win-64.

If i try to replicate the Savitzky-Golay example on Wikipedia, using `scipy.signal.savgol_coeffs` i get different results for the convolution coefficients.

Using: https://en.wikipedia.org/wiki/Savitzky%E2%80%93Golay_filter#Derivation_of_convolution_coefficients

For `window=5` and `order=4` Wikipedia gives:
`(1/35) * [ -3.  12.  17.  12.  -3.]`

And Scipy gives (`signal.savgol.coeffs(5,4)`):
`(1/35) * [ -1.261e-16   8.045e-15   3.500e+01   1.892e-14  -1.821e-15]`

I don't have access to the original publication Wiki references, it should contain pre-calculated tables. Here is one of those, which matches the Wiki example:
http://robert.mellet.pagesperso-orange.fr/rgrs_pol/regrs_06.htm

When i subtract 1 from the `order`, the result from Scipy matches Wiki and the website above, so there seems to be some incorrect shift in the order Scipy uses.

This also affects the `savgol_filter`, the weights Scipy uses for the window and order mentioned above basically take the center-value of the window, and thus the output signal is almost identical to the input signal. I'm no expert on the algorithm, but that doesn't seem right.

Here is a notebook i used for these experiments:
https://gist.github.com/RutgerK/091feca4f76730861dc78e879910f428

**Comment by Eric89GXL** [comment]
@WarrenWeckesser looks like you did most of this code. Do you have time to look?

**Comment by WarrenWeckesser** [comment]
The wikipedia example is "for smoothing by a 5-point **quadratic** polynomial" (emphasis added).  That corresonds to `polyorder=2`:

```
In [122]: savgol_coeffs(5, polyorder=2)
Out[122]: array([-0.08571429,  0.34285714,  0.48571429,  0.34285714, -0.08571429])

In [123]: savgol_coeffs(5, polyorder=2)*35
Out[123]: array([ -3.,  12.,  17.,  12.,  -3.])
```

**Comment by WarrenWeckesser** [comment]
The quote that I gave above is from the first example at the beginning of the wikipedia page: https://en.wikipedia.org/wiki/Savitzky%E2%80%93Golay_filter#Applications

**Comment by WarrenWeckesser** [comment]
This looks like a simple mistake ("quadratic" is order 2; order 4 would be called "quartic"), so I'm closing this issue.

@RutgerK, if you still think something is wrong, please comment, and if necessary, we can reopen the issue.

**Comment by RutgerK** [comment]
@WarrenWeckesser you're right, i misinterpreted quadratic. Sorry for the confusion.

**Comment by josef-pkt** [comment]
Just to agree with "confusing"

I always mix up what order and degree in polynomials mean.
I recently implemented something similar to savitzky-golay for statsmodels with tests against the scipy version and the two implementation agree in the overlapping functionality.
(However, I used a different literature and, ended up with a different definition of `order`. https://github.com/statsmodels/statsmodels/pull/3380#issuecomment-274339268 )
